comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -299,6 +347,45 @@ func (r *secretResolver) resolve(data []byte, origin string, notifyCb secrets.Re
 	return finalConfig, nil
 }
 
+// Refresh the secrets after they have been Resolved by fetching them from the backend again
+func (r *secretResolver) Refresh() error {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	newHandles := []string{}
+	for handle := range r.cache {
+		newHandles = append(newHandles, handle)
+	}
+
+	var secretResponse map[string]string
+	var err error
+	if r.fetchHookFunc != nil {
+		// hook used only for tests
+		secretResponse, err = r.fetchHookFunc(newHandles)
+	} else {
+		secretResponse, err = r.fetchSecret(newHandles)
+	}
+	if err != nil {
+		return err
+	}
+
+	for handle, secretValue := range secretResponse {
+		prevValue := r.cache[handle]
+		if prevValue == secretValue {
+			continue
+		}
+		for _, sub := range r.subscriptions {
+			contexts := r.origin[handle]
+			for _, ctx := range contexts {
+				sub(handle, ctx.path, prevValue, secretValue)
+			}
+		}
+		r.cache[handle] = secretValue
+	}
```

Comment:
This block is (mostly) duplicated above, it might make sense to move it to an helper function

Reply:
Done, now both call sites are using `handleSecretResponse`.